tgui/packages/tgui/interfaces/ExodroneConsole.tsx

```diff
@@ -0,0 +1,793 @@
+            average: [0.4 * drone_max_integrity, 0.7 * drone_max_integrity],
+            bad: [-Infinity, 0.4 * drone_max_integrity],
+          }}
+          value={drone_integrity}
+          maxValue={drone_max_integrity} />
+      </Stack.Item>
+    </Stack>);
+};
+
+const NoSiteDimmer = () => {
+  return (
+    <Dimmer>
+      <Stack textAlign="center" vertical>
+        <Stack.Item>
+          <Icon
+            color="red"
+            name="map"
+            size={10}
+          />
+        </Stack.Item>
+        <Stack.Item fontSize="18px" color="red">
+          No Destinations.
+        </Stack.Item>
+        <Stack.Item basis={0} color="red">
+          (Use the Scanner Array Console to find new locations.)
+        </Stack.Item>
+      </Stack>
+    </Dimmer>
+  );
+};
+
+const TravelTargetSelectionScreen = (props, context) => {
+  // List of sites and eta travel times to each
+  const { act, data } = useBackend<ExodroneConsoleData>(context);
+  const {
+    sites,
+    site,
+    can_travel,
+    travel_error,
+    drone_travel_coefficent,
+    all_bands,
+    drone_status,
+  } = data;
+
+  const travel_cost = target_site => {
+    if (site) {
+      return Math.max(Math.abs(site.distance - target_site.distance), 1) * drone_travel_coefficent;
+    }
+    else {
+      return target_site.distance * drone_travel_coefficent;
+    }
+  };
+  const [
+    choosingTools,
+    setChoosingTools,
+  ] = useLocalState(context, 'choosingTools', false);
+  const [
+    TravelDimmerShown,
+    setTravelDimmerShown,
+  ] = useLocalState(context, 'TravelDimmerShown', false);
+
+  const travel_to = ref => {
+    setTravelDimmerShown(false);
+    act("start_travel", { "target_site": ref });
+  };
+
+  return (
+    drone_status === "travel" && (
+      <TravelDimmer />
+    ) || (
+      <Section
+        title="Travel Destinations"
+        fill
+        scrollable
+        buttons={
+          <>
+            {props.showCancelButton && (
+              <Button
+                ml={5}
+                mr={0}
+                content="Cancel"
+                onClick={() => setTravelDimmerShown(false)} />
+            )}
+            <Box mt={props.showCancelButton && -3.5}>
+              <DroneStatus />
+            </Box>
+          </>
+        }>
+        {((sites && !sites.length) && !choosingTools) && (
+          <NoSiteDimmer />
+        )}
+        {site && (
+          <Section
+            mt={1}
+            title="Home"
+            buttons={
+              <Box>
+                <Button
+                  mr={1}
+                  content={can_travel ? "Launch!" : travel_error}
+                  onClick={() => travel_to(null)}
+                  disabled={!can_travel} />
+                ETA: {formatTime(site.distance * drone_travel_coefficent, "short")}
+              </Box>
+            }
+          />
+        )}
+        {sites.filter(destination => !site || destination.ref !== site.ref).map(destination => (
```

Comment:
honestly initially i gave this a pass but maybe this filter should be a constant on the function itself. i dunno i wouldn't be too beat up if you skipped